v0.13.0: panic in influxdb input plugin "interface is json.Delim, not string"

## Bug report

Telegraf process is sporadically crashing with `panic: interface conversion: interface is json.Delim, not string`. I'm pretty sure that none of the crash happened before update to 0.13.0. Around the same time, I have also updated InfluxDB to 0.13.0.

On the other hand, we have machines with the same Telegraf version (OS and everything) that haven't crashed since update.

### System info:

telegraf 0.13.0
OS: 64-bit Ubuntu Trusty 14.04

### Steps to reproduce:

Not consistently reproducible.

### Additional info:

I have a _feeling_ that this problem is happening under stressful conditions (like high CPU usage on the monitored host). See "Command timed out" for several minutes before..

```
2016/05/25 06:58:08 Gathered metrics, (1m0s interval), from 20 inputs in 8.949796949s
2016/05/25 06:58:19 Output [influxdb] buffer fullness: 99 / 10000 metrics. Total gathered metrics: 1040258. Total dropped metrics: 0.
2016/05/25 06:58:19 Output [influxdb] wrote batch of 99 metrics in 62.166429ms
2016/05/25 06:59:06 Error in input [exec]: exec: Command timed out. for command '/usr/lib/telegraf/scripts/sumproc.sh -u sentry'
2016/05/25 06:59:07 Error in input [exec]: exec: Command timed out. for command '/usr/lib/telegraf/scripts/sumproc.sh -f /var/run/icinga2/icinga2.pid'
2016/05/25 06:59:07 Error in input [exec]: exec: Command timed out. for command '/usr/lib/telegraf/scripts/sumproc.sh -f /var/run/postgresql/9.3-main.pid'
2016/05/25 06:59:07 Error in input [exec]: exec: Command timed out. for command '/usr/lib/telegraf/scripts/sumproc.sh -f /var/run/telegraf/telegraf.pid'
2016/05/25 06:59:07 Gathered metrics, (1m0s interval), from 20 inputs in 7.459498165s
2016/05/25 06:59:24 Output [influxdb] buffer fullness: 97 / 10000 metrics. Total gathered metrics: 1040355. Total dropped metrics: 0.
2016/05/25 06:59:24 Output [influxdb] wrote batch of 97 metrics in 156.915989ms
panic: interface conversion: interface is json.Delim, not string

goroutine 449482 [running]:
panic(0x10834e0, 0xc820677d80)
        /usr/local/go/src/runtime/panic.go:481 +0x3e6
github.com/influxdata/telegraf/plugins/inputs/influxdb.(*InfluxDB).gatherURL(0xc8200c8440, 0x7f05202373f8, 0xc8204ef280, 0xc8201bb621, 0x20, 0x0, 0x0)
        /root/go/src/github.com/influxdata/telegraf/plugins/inputs/influxdb/influxdb.go:157 +0x328
github.com/influxdata/telegraf/plugins/inputs/influxdb.(*InfluxDB).Gather.func1(0xc8201db340, 0xc8200c8440, 0x7f05202373f8, 0xc8204ef280, 0xc82004d800, 0xc8201bb621, 0x20)
        /root/go/src/github.com/influxdata/telegraf/plugins/inputs/influxdb/influxdb.go:45 +0xa5
created by github.com/influxdata/telegraf/plugins/inputs/influxdb.(*InfluxDB).Gather
        /root/go/src/github.com/influxdata/telegraf/plugins/inputs/influxdb/influxdb.go:48 +0x185
2016/05/25 08:44:19 Starting Telegraf (version 0.13.0)
2016/05/25 08:44:19 Loaded outputs: influxdb
2016/05/25 08:44:19 Loaded inputs: processes exec swap system cpu kernel mem net netstat disk diskio exec exec exec influxdb exec postgresql exec redis exec
2016/05/25 08:44:19 Tags enabled: Name=monitoring host=ip-10-73-170-141
2016/05/25 08:44:19 Agent Config: Interval:1m0s, Debug:false, Quiet:false, Hostname:"ip-10-73-170-141", Flush Interval:1m0.371046598s 
```

Reading the stacktrace, it is possible that the problem is in `influxdb` plugin. This machine is monitoring the same InfluxDB, which is also the target for the metrics. Telegraf config snippet..
```
[agent]
  interval = "60s"
  flush_interval = "60s"
  flush_jitter = "10s"

[[outputs.influxdb]]
  urls = ["http://xxxx:8086"]
  database = "telegraf"
  precision = "s"

[[inputs.influxdb]]
  urls = [ "http://127.0.0.1:8086/debug/vars", ]
```
Monitored InfluxDB is also on 0.13.0.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -27,6 +27,7 @@ time before a new metric is included by the plugin.
 - [#1252](https://github.com/influxdata/telegraf/pull/1252): Fix systemd service. Thanks @zbindenren!
 - [#1221](https://github.com/influxdata/telegraf/pull/1221): Fix influxdb n_shards counter.
 - [#1258](https://github.com/influxdata/telegraf/pull/1258): Fix potential kernel plugin integer parse error.
+- [#1268](https://github.com/influxdata/telegraf/pull/1268): Fix potential influxdb input type assertion panic.
 
 ## v0.13.1 [2016-05-24]
 

--- a/plugins/inputs/influxdb/influxdb.go
+++ b/plugins/inputs/influxdb/influxdb.go
@@ -157,43 +157,45 @@ func (i *InfluxDB) gatherURL(
 			return err
 		}
 
-		if key.(string) == "memstats" {
-			var m memstats
-			if err := dec.Decode(&m); err != nil {
-				continue
+		if keyStr, ok := key.(string); ok {
+			if keyStr == "memstats" {
+				var m memstats
+				if err := dec.Decode(&m); err != nil {
+					continue
+				}
+				acc.AddFields("influxdb_memstats",
+					map[string]interface{}{
+						"alloc":           m.Alloc,
+						"total_alloc":     m.TotalAlloc,
+						"sys":             m.Sys,
+						"lookups":         m.Lookups,
+						"mallocs":         m.Mallocs,
+						"frees":           m.Frees,
+						"heap_alloc":      m.HeapAlloc,
+						"heap_sys":        m.HeapSys,
+						"heap_idle":       m.HeapIdle,
+						"heap_inuse":      m.HeapInuse,
+						"heap_released":   m.HeapReleased,
+						"heap_objects":    m.HeapObjects,
+						"stack_inuse":     m.StackInuse,
+						"stack_sys":       m.StackSys,
+						"mspan_inuse":     m.MSpanInuse,
+						"mspan_sys":       m.MSpanSys,
+						"mcache_inuse":    m.MCacheInuse,
+						"mcache_sys":      m.MCacheSys,
+						"buck_hash_sys":   m.BuckHashSys,
+						"gc_sys":          m.GCSys,
+						"other_sys":       m.OtherSys,
+						"next_gc":         m.NextGC,
+						"last_gc":         m.LastGC,
+						"pause_total_ns":  m.PauseTotalNs,
+						"num_gc":          m.NumGC,
+						"gcc_pu_fraction": m.GCCPUFraction,
+					},
+					map[string]string{
+						"url": url,
+					})
 			}
-			acc.AddFields("influxdb_memstats",
-				map[string]interface{}{
-					"alloc":           m.Alloc,
-					"total_alloc":     m.TotalAlloc,
-					"sys":             m.Sys,
-					"lookups":         m.Lookups,
-					"mallocs":         m.Mallocs,
-					"frees":           m.Frees,
-					"heap_alloc":      m.HeapAlloc,
-					"heap_sys":        m.HeapSys,
-					"heap_idle":       m.HeapIdle,
-					"heap_inuse":      m.HeapInuse,
-					"heap_released":   m.HeapReleased,
-					"heap_objects":    m.HeapObjects,
-					"stack_inuse":     m.StackInuse,
-					"stack_sys":       m.StackSys,
-					"mspan_inuse":     m.MSpanInuse,
-					"mspan_sys":       m.MSpanSys,
-					"mcache_inuse":    m.MCacheInuse,
-					"mcache_sys":      m.MCacheSys,
-					"buck_hash_sys":   m.BuckHashSys,
-					"gc_sys":          m.GCSys,
-					"other_sys":       m.OtherSys,
-					"next_gc":         m.NextGC,
-					"last_gc":         m.LastGC,
-					"pause_total_ns":  m.PauseTotalNs,
-					"num_gc":          m.NumGC,
-					"gcc_pu_fraction": m.GCCPUFraction,
-				},
-				map[string]string{
-					"url": url,
-				})
 		}
 
 		// Attempt to parse a whole object into a point.

